### PR TITLE
My Jetpack: Pass slug prop to ProductCard

### PR DESCRIPTION
**Pull request by oskosk**
Needed for #22755

#### Changes proposed in this Pull Request:

* Makes `slug` be a required Prop in ProductCard
* Makes the `useProduct` hook return `slug`
* Makes the specific product components pass slug on <ProductCard/> usage.

#### Testing instructions:

* Visit My Jetpack.
* Confirm you see no errors in the console (Due to slug now being a reuquired prop)

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/components/product-cards-section/anti-spam-card.jsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const AntiSpamIcon = () => (
 );
 
 const AntiSpamCard = ( { admin } ) => {
-	const { status, activate, deactivate, detail, isFetching } = useProduct( 'anti-spam' );
+	const { slug, status, activate, deactivate, detail, isFetching } = useProduct( 'anti-spam' );
 	const { name, description } = detail;
```
Comment (retrofox) [inline projects/packages/my-jetpack/_inc/components/product-cards-section/anti-spam-card.jsx:26]:
```suggestion
	const { name, description, slug } = detail;
```

Comment (retrofox) [inline projects/packages/my-jetpack/_inc/components/product-cards-section/anti-spam-card.jsx:26]:
```suggestion
	const { status, activate, deactivate, detail, isFetching } = useProduct( 'anti-spam' );
	const { name, description, slug } = detail;
```

Comment (oskosk) [reply projects/packages/my-jetpack/_inc/components/product-cards-section/anti-spam-card.jsx:26]: oh thanks

Comment (oskosk) [reply projects/packages/my-jetpack/_inc/components/product-cards-section/anti-spam-card.jsx:26]: > The idea behind using the detail object is to collect all product data relevant to it while exposing helpers and not-product data in the root. Thus

Is `slug` a "detail" though? It's more the key we use to access the detailed information. So we actually know it in advance when calling `useProduct()`. Anyways, working on a commit to consume it from `detail`.

Comment (oskosk) [reply projects/packages/my-jetpack/_inc/components/product-cards-section/anti-spam-card.jsx]: Addressed in [a40e2b2](https://github.com/Automattic/jetpack/pull/22756/commits/a40e2b2e0af25e1aad67f10d94d578e4cdd0bfe3)

**Comment by oskosk** [comment]
Addressed tests in 117c5b918608886946647d828e09d646678a9c4f. @retrofox could you take a look again ?

**Review by retrofox — APPROVED** [review]
LGTM

**Comment by oskosk** [comment]
Thanks!

**Pull request merged**